component/web/security/src/main/java/org/exoplatform/web/login/LoginServlet.java

```diff
@@ -287,7 +287,8 @@ public String getContextPath() {
 
         try {
             URI uri = new URI(initialURI);
-            if (uri.isAbsolute() && !(uri.getHost().equals(req.getServerName()))) {
+            // sanitizing url from suspicious uri
+            if (!uri.getPath().equals(req.getContextPath())) {
```

Comment:
We can check just for the server name in both cases : 
```suggestion
            if (!(uri.getHost().equals(req.getServerName()))) {
```

Reply:
uri.getHost() is returning null and when evaluating the expression the server returns a null pointer exception . otherwise the standard URL of the login page has the value of null for the uri so with the expression you suggested we would always have  to force setting the uri to "/portal " for all the cases which i think is not recommended.